### PR TITLE
register batchnorm for opset14 cuda provider

### PR DESCRIPTION
**Pull request by askhade**
**Description**: Register batch norm opset 14 for cuda provider

**Motivation and Context**
- Enable complete support for onnx ops in ort
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Review by SherlockNoMad — APPROVED** [review]

File: onnxruntime/core/providers/cuda/nn/batch_norm.cc [hunk 1]
```diff
@@ -21,10 +21,19 @@ namespace cuda {
...
+      T,                                                          \
+      kCudaExecutionProvider,                                     \
+      (*KernelDefBuilder::Create())                               \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      BatchNorm<T>);                                              \
   ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
       BatchNormalization,                                         \
       kOnnxDomain,                                                \
-      9,                                                          \
+      14,                                                         \
```
Comment (hariharans29) [inline onnxruntime/core/providers/cuda/nn/batch_norm.cc:36]: It looks like this kernel registration has a small bug. Opset-14 introduced a new type constraint 'U' for mean/var. The implementation only supports cases where T == U. For T != U cases, it will crash within the kernel when we reference data of type T for the inputs containing type U. I guess we need to add `.TypeConstraint("U",DataTypeImpl::GetTensorType<T>())` here.
